### ISSUE TITLE
add_torrent "download_location" not taken into account

### ISSUE THREAD

**Issue by madchap** [idlesign/deluge-webapi#11]
Hi,

Thanks for the plugin!

When doing an add_torrent, the "download_location" is not taken into consideration. I am well logged in, and passing cookies and so on. The torrent actually gets added to the link, but the location is always the default one that is set in the webUI preferences.

I am running Deluge 1.3.15.

```
torrent_payload = {
            "id": 1,
            "method": "webapi.add_torrent",
            "params": [magnet, {"download_location": "/mnt/plexdata/Series/"}]
        }
r = requests.post(json_ep, json=torrent_payload, cookies=deluge_cookie)
```

![image](https://user-images.githubusercontent.com/5468769/40452088-f68c3ffa-5ee0-11e8-8b90-a231f824fb3a.png)

Thanks for your hints.

**Comment by idlesign** [comment]
Hi, thank you for the information.

Can't say for sure, as I don't use Deluge these days. 
But you can check whether the issue is somehow related to the usage of `Move Completed` option https://dev.deluge-torrent.org/ticket/2012

If you don't use the said option, please let me know, I'll try to make some time for the investigation.

**Comment by madchap** [comment]
Hi,

Thanks for the prompt answer.

No, I do not use that option at all.

Let me know if I can help. If I find time, I'll try to dig a bit in the code too (I am no wizard however). If you have specific pointers as to where to start looking, I'll gladly take them :)

**Comment by idlesign** [comment]
You could turn on verbose/debug mode for deluge, and see in logs whether this option is at all received by and digested by deluge.

**Comment by madchap** [comment]
Hi,

So putting in deluge-web in debug mode, seems like it is indeed not honored.

My requests
```
May 24 20:49:06 helmsdeep.home deluge-web[20327]: [DEBUG   ] 20:49:06 json_api:270 json-request: {"id": 1, "method": "auth.login", "params": ["xxxxxxxxxx"]}
May 24 20:49:06 helmsdeep.home deluge-web[20327]: [DEBUG   ] 20:49:06 auth:199 Received a password via the 1.2 auth method
May 24 20:49:06 helmsdeep.home deluge-web[20327]: [INFO    ] 20:49:06 auth:327 Login success (ClientIP 192.168.10.254)
May 24 20:49:06 helmsdeep.home deluge-web[20327]: [DEBUG   ] 20:49:06 auth:146 Creating session for admin
May 24 20:49:06 helmsdeep.home deluge-web[20327]: [DEBUG   ] 20:49:06 json_api:270 json-request: {"id": 1, "method": "auth.check_session", "params": []}
May 24 20:49:06 helmsdeep.home deluge-web[20327]: [DEBUG   ] 20:49:06 json_api:270 json-request: {"id": 1, "method": "webapi.add_torrent", "params": ["magnet:?xt=....announce", {"download_location": "/mnt/plexdata/Series/"}]}
```

What it does with it.
```
May 24 20:49:06 helmsdeep.home deluge-web[20327]: [DEBUG   ] 20:49:06 json_api:270 json-request: {"id": 1, "method": "webapi.add_torrent", "params": ["magnet:?xt=....announce` using options `{u'download_location': '/mnt/plexdata/'}` ...
```

**Comment by idlesign** [comment]
Hi,

That's interesting. What's `/mnt/plexdata/`? Is it a computed default or it's from a configuration file?
Since some people are using 1.3.15 without such an issue, maybe it's somehow related to deamon or web configuraion.

**Comment by madchap** [comment]
It is the default setting for `download_location` that I set in the webUI.

![image](https://user-images.githubusercontent.com/5468769/40530956-f6e568da-5ffa-11e8-9b2a-6c905a7ec27f.png)

**Comment by idlesign** [comment]
I see now, that's #8. I've just forgot to release version 0.2.1, with the fix, sorry.
Will try to do it in a couple of days.

**Comment by idlesign** [comment]
Just released 0.2.1. You can give it a try.

**Comment by madchap** [comment]
Indeed fixed with 0.2.1! Thanks :+1:
